sphinx/ext/autosummary/generate.py

```diff
@@ -430,8 +430,7 @@ def find_autosummary_in_lines(lines: List[str], module: str = None, filename: st
             if m:
                 toctree = m.group(1)
                 if filename:
-                    toctree = os.path.join(os.path.dirname(filename),
-                                           toctree)
+                    _, toctree = app.env.relfn2path(toctree, filename)
```

Comment:
Please let me know what are you doing in this line. The 2nd parameter for `relfn2path()` is a docname, not a filename. I also don't understand why do you want to change this to an absolute path. What happens in this change?

Reply:
Well, if I understood the code, the `dirname` of the second argument is used as the starting point of relative paths. I tried to preserve the behaviour of relative paths, while making absolute paths relative to the doc folder.

Sorry, because this is the first time that I am trying to contribute to this project and I do not know the terminology. What is the difference between docname and filename in this context?

Reply:
A docname is an abbreviation of "document name". And it is a logical name for the document, like a relative path string from source directory without file-extension.
https://www.sphinx-doc.org/en/master/glossary.html#glossary

I know the method is almost same as you'd like to do. But it is not appropriate method for this purpose. The `generate_autosummary_docs()` is used in both `sphinx-build` and apidoc. And apidoc does not know source directory and "document".

BTW, I still don't understand what will be changed by this line. In my understanding correct, the `filename` is an absolute path under the source directory (in sphinx-build command). So I guess this does not change the behavior, right?

Reply:
It will change absolute paths prepending the doc folder.